web/html/src/components/input/Form.js

```diff
@@ -76,14 +76,18 @@ export class Form extends React.Component<Props> {
   }
 
   unregisterInput(component: React.ElementRef<any>) {
-    if (component.props && component.props.name && this.inputs[component.props.name] === component) {
-      delete this.inputs[component.props.name];
+    if (component.props && component.props.name){
+      const name = component.props.name instanceof Array ? component.props.name.join() : component.props.name;
+      if (this.inputs[name] === component) {
+        delete this.inputs[component.props.name];
+      }
     }
   }
 
   registerInput(component: React.ElementRef<any>) {
     if (component.props && component.props.name) {
-      this.inputs[component.props.name] = component;
+      const name = component.props.name instanceof Array ? component.props.name.join() : component.props.name;
```

Comment:
I'm not sure I understand, given an array of names, what's the difference between registering the components as a joint string vs registering all of them separately? I think I simply don't understand the intent here, a comment could help a lot.

Reply:
if we register them all separately we'll have validates for each property instead of having one with all properties of the field... and I wanted to avoid touching that logic too heavily

Reply:
I'm not familiar with the validator logic, so perhaps @ncounter or @cbbayburt can chime in here as well — I'm not sure which will be simpler, adding array names support (this current PR) or updating the validator logic.  

I haven't touched the forms myself so I don't know too much about their usage. Simply as food for though I would offer an alternative consideration that it would also be an option to keep inputs simpler and have the complexity in the validation layer, something along the lines of:  

```jsx
<DateTimePicker name="fooStart" validateAs="fooRange" />
<DateTimePicker name="fooEnd" validateAs="fooRange" />
```

I'm mostly throwing around ideas here though, I think the other guys will have much better informed opinions on the matter.